tests/geomath_tests.rs

```diff
@@ -0,0 +1,244 @@
+// Integration tests related to geomath.
+
+extern crate utilities;
+
+use geographiclib_rs::geomath;
+use utilities::{assert_delta, test_basic};
+
+// reminder: use "cargo test -- --nocapture" to let tests print to console.
+// reminder: if a test fails especially badly, review the C++ "instrumented-crude" logging for possible bugs there
+
+// todo: decide between keeping these as integration tests vs unit tests vs mixed...
+//       why integration: external files and parsing code (not self-contained)
+//       why unit: sometimes want access to private/internal data (e.g. consts, constructors)
+
+#[test]
+fn test_ang_diff_vs_cpp() {
+    // Format: x y result=z e-out
+    test_basic("Math_AngDiff_x_y_e", 4, |line_num, items| {
+        let result = geomath::ang_diff(items[0], items[1]);
+        // todo: try to pass allow_sign_change=false
+        assert_delta!(items[2], result.0, 0.0, true, "result (result.0)", line_num);
+        // todo: try to pass allow_sign_change=false
+        assert_delta!(items[3], result.1, 0.0, true, "e (result.1)", line_num);
+    });
+}
+
+#[test]
+fn test_ang_normalize_vs_cpp() {
+    // Format: x result
+    test_basic("Math_AngNormalize", 2, |line_num, items| {
+        let result = geomath::ang_normalize(items[0]);
+        // todo: try to pass allow_sign_change=false
+        assert_delta!(items[1], result, 0.0, true, "result", line_num);
+    });
+}
+
+#[test]
+fn test_ang_round_vs_cpp() {
+    // Format: x result
+    test_basic("Math_AngRound", 2, |line_num, items| {
+        let result = geomath::ang_round(items[0]);
+        assert_delta!(items[1], result, 0.0, false, "result", line_num);
+    });
+}
+
+#[test]
+fn test_atan2d_vs_cpp() {
+    // Format: y x result
+    test_basic("Math_atan2d", 3, |line_num, items| {
+        let result = geomath::atan2d(items[0], items[1]);
+        // todo: try to reduce allowed difference to 0.0
+        assert_delta!(items[2], result, 5e-16, false, "result", line_num);
+    });
+}
+
+// #[test]
+// fn test_atand_vs_cpp() {
+//     // Format: x result
+//     test_basic("Math_atand", 2, |line_num, items| {
+//         let result = geomath::atand(items[0]);
+//         assert_delta!(items[1], result, 0.0, false, "result", line_num);
+//     });
+// }
+
+#[test]
+fn test_consts_vs_cpp() {
+    // Format: digits digits10 extra_digits bigendian pi degree GEOGRAPHICLIB_PRECISION GEOGRAPHICLIB_WORDS_BIGENDIAN
+    // todo: review commented-out tests to see if there's an equivalent const or simple calculation we can compare
+    let items = utilities::read_consts_basic("Math_consts", 8);
+    let line_num = 2; // todo: remove once assert_delta stops requiring a line number
+    assert_delta!(items[0], geomath::DIGITS as f64, 0.0, false, "DIGITS", line_num);
+    // assert_delta!(items[1], geomath::DIGITS10 as f64, 0.0, false, "DIGITS10", line_num);
```

Comment:
Most of those are commented out because they would require access to non-public values in the current Rust implementation, and that's not available to the current integration-test formulation of the test. Some of the constants may also not exist in the Rust version -- I haven't dug to make that distinction.